sklearn/cross_decomposition/tests/test_pls.py

```diff
@@ -383,52 +383,65 @@ def test_copy(Est):
 
 
 @pytest.mark.parametrize('Est', (CCA, PLSCanonical, PLSRegression, PLSSVD))
-def test_scale_and_stability(Est):
+def test_scale_and_stability_non_regression_7819(Est):
+    # Non-regression for https://github.com/scikit-learn/scikit-learn/pull/7819
+    rng = np.random.RandomState(0)
+
+    n_samples = 1000
+    n_targets = 5
+    n_features = 10
+    Q = rng.randn(n_targets, n_features)
+    Y = rng.randn(n_samples, n_targets)
+    X = np.dot(Y, Q) + 2 * rng.randn(n_samples, n_features) + 1
+    X *= 1000
+
+    X_s, Y_s, *_ = _center_scale_xy(X, Y)
+
+    X_score, Y_score = Est(scale=True).fit_transform(X, Y)
+    X_s_score, Y_s_score = Est(scale=False).fit_transform(X_s, Y_s)
+
+    assert_array_almost_equal(X_s_score, X_score)
+    assert_array_almost_equal(Y_s_score, Y_score)
+
+
+@pytest.mark.parametrize('Est', (CCA, PLSCanonical, PLSRegression, PLSSVD))
+def test_scale_and_stability_non_regression_linnerud(Est):
     # scale=True is equivalent to scale=False on centered/scaled data
     # This allows to check numerical stability over platforms as well
 
-    rng = np.random.RandomState(0)
-
-    d = load_linnerud()
-    X1 = d.data
-    Y1 = d.target
+    X, Y = load_linnerud(return_X_y=True)
     # causes X[:, -1].std() to be zero
-    X1[:, -1] = 1.0
+    X[:, -1] = 1.0
+    X_s, Y_s, *_ = _center_scale_xy(X, Y)
+
+    X_score, Y_score = Est(scale=True).fit_transform(X, Y)
+    X_s_score, Y_s_score = Est(scale=False).fit_transform(X_s, Y_s)
 
+    assert_array_almost_equal(X_s_score, X_score)
+    assert_array_almost_equal(Y_s_score, Y_score)
+
+
+@pytest.mark.parametrize('Est', (CCA, PLSCanonical, PLSRegression, PLSSVD))
+def test_scale_and_stability_2821(Est):
     # From bug #2821
     # Test with X2, Y2 s.t. clf.x_score[:, 1] == 0, clf.y_score[:, 1] == 0
```

Comment:
BTW I think this comment is wrong / outdated. I could not observe y_score[:, 1] to be zero, same for x_score.